pkg/segment/aggregations/evalaggs.go

```diff
@@ -693,6 +721,62 @@ func ComputeAggEvalForValues(measureAgg *structs.MeasureAggregator, sstMap map[s
 	return nil
 }
 
+func ComputeAggEvalForList(measureAgg *structs.MeasureAggregator, sstMap map[string]*structs.SegStats, measureResults map[string]utils.CValueEnclosure, runningEvalStats map[string]interface{}) error {
+	fields := measureAgg.ValueColRequest.GetFields()
+	finalList := []string{}
+
+	var strList []string
+	_, ok := runningEvalStats[measureAgg.String()]
+	if !ok {
+		strList = make([]string, 0)
+		runningEvalStats[measureAgg.String()] = strList
+	} else {
+		strList, ok = runningEvalStats[measureAgg.String()].([]string)
+		if !ok {
+			return fmt.Errorf("ComputeAggEvalForList: can not convert to list for measureAgg: %v", measureAgg.String())
+		}
+	}
+
+	if len(fields) == 0 {
+		fieldToValue := make(map[string]utils.CValueEnclosure)
+		list, err := PerformAggEvalForList(measureAgg, strList, fieldToValue)
+		if err != nil {
+			return fmt.Errorf("ComputeAggEvalForList: Error while performing eval agg for list, err: %v", err)
+		}
+		finalList = list
+	} else {
+		sst, ok := sstMap[fields[0]]
+		if !ok {
+			return fmt.Errorf("ComputeAggEvalForList: sstMap did not have segstats for field %v, measureAgg: %v", fields[0], measureAgg.String())
+		}
+
+		length := len(sst.Records)
+		for i := 0; i < length; i++ {
```

Comment:
nit: this is more idiomatic for Go
```suggestion
		for i := range sst.Records {
```

Reply:
Switched to range loop